src/main/java/org/opensearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -228,16 +228,18 @@ public void run() {
 
     public void shutdown() {
         try {
-            conn.close();
+            if (!conn.isClosed()) {
+                conn.close();
+            }
         } catch (Exception e) {
-            LOG.error("Unable to close inmemory database connection.");
+            LOG.error("Unable to close inmemory database connection.", e);
```

Comment:
Can emit metrics in case we are not able to close connections?

Reply:
Yeah makes sense. Added.

Reply:
Added.